src/internal-model.js

```diff
@@ -118,7 +129,13 @@ export default Model.extend({
    * @returns {Boolean} true if relationship exists
    */
   hasRelated (relationName) {
-    return !_.isEmpty(this.getRelationshipData(relationName, false))
+    const relatedData = this.getRelationshipData(relationName, false)
+
+    if (_.isEmpty(relatedData)) return false
+
+    if (_.isArray(relatedData)) return _.reduce(relatedData, (acc, data) => acc && isValidResourceIdentifier(data), true)
```

Comment:
FYI this is `every`

Reply:
```suggestion
    if (_.isArray(relatedData)) return _.every(relatedData, isValidResourceIdentifier)
```
Oh, that is so much nicer.

Reply:
Also native, for those who care:https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every

Reply:
```suggestion
  return _.isArray(relatedData)
      ? relatedData.every(isValidResourceIdentifier)
      : isValidResourceIdentifier(relatedData)
```

😭

Reply:
```suggestion
    return [].concat(relatedData).every(isValidResourceIdentifier);
```
I can play this game all day.

Reply:
```suggestion
    return !_.isEmpty(relatedData) && [].concat(relatedData).every(isValidResourceIdentifier)
```
LOL. Need to take that extra bit of time to optimize and simplify more often.

Reply:
Actually, you can drop the `isEmpty` predicate into `isValidResourceIdentifier`, right?

Reply:
I don't think that would work because `every` would not call `isValidResourceIdentifier` with an empty array. `relatedData` could be an empty array itself.